waffle-chai/test/matchers/eventsTest.ts

```diff
@@ -698,5 +699,43 @@ export const eventsWithNamedArgs = (provider: TestProvider) => {
         '{ Object (hash, value, ...) } to deeply equal { Object (hash, value, ...) }'
       );
     });
+
+    it('Signature only - delegatecall', async () => {
+      const proxyFactory = new ContractFactory(EventsProxyABI.abi, EventsProxyABI.bytecode, wallet);
+      const proxy = await proxyFactory.deploy(events.address);
+
+      await expect(proxy.emitTwoDelegate()).to.emit('Two(uint256,string)');
```

Comment:
There is one more scenario - transaction on contract A calls a function on contract B and contract B emits an event. We want to be able to capture this event as well.